case-lib/pipeline.sh

```diff
@@ -18,7 +18,7 @@ func_pipeline_export()
         for key in "${!TPLG_IGNORE_LST[@]}"
         do
             dlogi "Pipeline list to ignore is specified, will ignore '$key=${TPLG_IGNORE_LST[$key]}' in test case"
-            ignore=$ignore" $key:${TPLG_IGNORE_LST[$key]}"
+            ignore=$ignore"$key:${TPLG_IGNORE_LST[$key]};"
```

Comment:
The commit message states "use semicolon to comply with semantics..." Are these semantics new and if yes then since when? If there are not new then was this code buggy and  not working? Is this a fix? Maybe some ignore example(s) in the commit message would help.

I can't understand the new location of the delimiter either.

Please quote `"$ignore"` too, at least for consistency.

Reply:
Semantics is new, and since this PR, commit id is 9c2750405873dc8c178c91201168b89919f3a6da. I will explain more.
> The filtering feature now allows us to acquire specific pipelines we want,  for example, pipelines with/without PGA/EQ/ASRC/KPBM/ECHO REF/SMART_AMP component, pipelines of specific IDs and pipelines of playback/capture. 

>  Recently, Echo Reference feature is introduced, this feature requires two pipelines, the capture pipeline will capture the data of the playback pipeline, you can regard this as loop back. with the filtering feature, `-f "echo:any" ` will help to acquire the two pipelines as they both contain ECHO REF compnent, and it will be easy to put this case to CI.

>  Our capture test case use -f "type:capture" to acquire all capture pipelines, and the capture from Echo Reference is also here. Here comes the problem, Echo Reference capture cannot work alone, all of our capture case will fail due to this pipeline.

> This PR is intended to solve the above problem. The block feature allow us to block a single pipeline. `-b type:capture & echo:any` will do. We also need to block WOV pipeline in capture test, that's where the `delimiter ";"` works. `-b type:capture & echo:any; kpbm:any`

Reply:
@marc-hb As the block feature is enhanced in this PR, the test script do need a enhance to use the new feature. Let's do this  in another PR to make changes consistent. So this PR, only the enhancement on sof-tplgreader.py. this enhancement is required to add echo reference case to CI.